Syntaxes/SCSS.sublime-syntax

```diff
@@ -7,379 +7,3623 @@ file_extensions:
 scope: source.scss
 
 variables:
-  unicode: '\\\h{1,6}[ \t\n\f]?'
-  escape: '(?:{{unicode}}|\\[^\n\f\h])'
-  nonascii: '[\p{L}\p{M}\p{S}\p{N}&&[^[:ascii:]]]'
-  nmstart: '(?:[[_a-zA-Z]{{nonascii}}]|{{escape}})'
-  nmchar: '(?:[[-\w]{{nonascii}}]|{{escape}})'
-  ident: '(?:--{{nmchar}}+|-?{{nmstart}}{{nmchar}}*)'
+  # Basic tokens
+  # https://www.w3.org/TR/css3-selectors/#lex
+  unicode: \\\h{1,6}[ \t\n\f]?
+  escape: (?:{{unicode}}|\\[^\n\f\h])
+  nmstart: (?:[_[:alpha:][:^ascii:]]|{{escape}})
+  nmchar: (?:[-_[:alnum:][:^ascii:]]|{{escape}})
+
+  # Identifier Break
+  # The proper pattern would be (?!{{nmchar}}), but its impact on performance
+  # is just too high, thus several optimizations are applied.
+  # 1. Use positive lookahead with \Z to handle `eof`
+  # 2. Breaks are ascii characters other than denoted by {{nmchar}}.
+  # 3. Assume unicode or escape character if backslash is matched, instead of
+  #    matching (?!{{escape}}) which also effects performance negative.
+  break: (?=[[^-_[:alnum:]\\]&&[[:ascii:]]]|\Z)
+
+  # Identifiers
+  # https://www.w3.org/TR/css-syntax-3/#typedef-ident-token
+  # https://www.w3.org/TR/css3-selectors/#lex
+  ident: (?:{{custom_ident}}|{{generic_ident}})
+  custom_ident: (?:--{{nmchar}}*)
+  generic_ident: (?:-?{{nmstart}}{{nmchar}}*)
+
+  # Ilegal Custom Identifier Names
+  # https://www.w3.org/TR/css-values-4/#custom-idents
+  illegal_custom_ident: |-
+    \b{{illegal_custom_ident_tokens}}{{break}}
+  illegal_custom_ident_tokens: |-
+    (?xi: auto | default | inherit | initial | none | unset )
+
+  # Types
+  # https://www.w3.org/TR/css3-values/#numeric-types
+  # https://www.w3.org/TR/css-syntax-3/#number-token-diagram
+  integer: ([-+]?)(\d+)
+  float: ([-+]?)(\d*(\.)\d+(?:[eE][-+]?\d+)?|\d+[eE][-+]?\d+)
+
+  # Units
+  # https://www.w3.org/TR/css3-values/#lengths
+  units: |-
+    (?x:
+      %
+    | {{absolute_lengths}}
+    | {{angle_units}}
+    | {{duration_units}}
+    | {{font_relative_lengths}}
+    | {{frequency_units}}
+    | {{resolution_units}}
+    | {{viewport_percentage_lengths}}
+    )
+  font_relative_lengths: (?i:em|ex|ch|rem)\b
+  viewport_percentage_lengths: (?i:vw|vh|vmin|vmax)\b
+  absolute_lengths: (?i:cm|mm|q|in|pt|pc|px|fr)\b
+  angle_units: (?i:deg|grad|rad|turn)\b
+  duration_units: (?i:s|ms)\b
+  frequency_units: (?i:Hz|kHz)\b
+  resolution_units: (?i:dpi|dpcm|dppx)\b
+
+  # Combinators
+  # https://drafts.csswg.org/selectors-4/#combinators
+  combinators: (?:>{1,3}|[~+]|\|{2})
+
+  vendor_prefix: -(?:webkit|moz|ms|o)-
+
+  # Custom Element Names
+  # https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-core-concepts
+  # adjusted from CSS so it doesn't match on interpolation (#{)
+  custom_element_tags: \b[a-z]{{custom_element_chars}}*-{{custom_element_chars}}*(?!#{){{break}}
+  # Note: Excludes `.` as it is used to identify attribute access
+  custom_element_chars: |-
```

Comment:
You might be able to use <https://github.com/kkos/oniguruma/blob/master/doc/UNICODE_PROPERTIES> via `\p{PROPERY_NAME}` for some of these.